mir-ci/mir_ci/robot_resources/screencopy.resource

```diff
@@ -8,93 +9,161 @@ Library             ../robot_libraries/WaylandHid.py
     ELSE
         ${position}=    Set Variable    ${destination}
     END
-    WaylandHid.Move Pointer To Absolute    ${position}[x]    ${position}[y]
+    WaylandHid.Move Pointer To Absolute    ${position}[0]    ${position}[1]
     RETURN    ${position}
 
-Move Pointer To Proportional ${position}
-    [Documentation]    Move the pointer to a position given as proportions to
-    ...    the size of the display output.
+Move Pointer To Proportional (${x}, ${y})
+    [Documentation]    Move the pointer to a destination position given as
+    ...    proportions to the size of the display output.
     ...
-    ...    ${position} must be a point represented by floating-point x and y
-    ...    values in the range 0..1.
+    ...    Arguments:
+    ...    - ${x}: Relative x-coordinate to move the pointer to. It must be in
+    ...      the range 0..1, where 0 represents the left edge, and 1 represents
+    ...      the right edge of the output.
+    ...    - ${y}: Relative y-coordinate to move the pointer to. It must be in
+    ...      the range 0..1, where 0 represents the top edge, and 1 represents
+    ...      the bottom edge of the output.
     ...
-    ...    The return value is the pointer position after the move,
-    ...    in absolute coordinates.
-    WaylandHid.Move Pointer To Proportional    ${position}[x]    ${position}[y]
+    ...    Return:
+    ...    Absolute position of the pointer after the move, as a tuple (x, y)
+    ...    of integers.
+    ${position}=    WaylandHid.Move Pointer To Proportional    ${x}    ${y}
     RETURN    ${position}
 
-Walk Pointer From ${start_position} To ${destination}
-    [Documentation]    Move the pointer in incremental steps from a start
-    ...    position to a destination position or image template.
-    ...
-    ...    When ${destination} is the path of an image template file, the
-    ...    corresponding position will be that of the center of the first
-    ...    matching template region.
-    ...
-    ...    Optional arguments:
-    ...    ${steps}: number of steps. If <= 0, the pointer is moved in
-    ...    increments of 16 pixels. Default is 0.
-    ...    ${delay}: time to sleep after each step, in seconds.
-    ...    Default is 0.
-    ...
-    ...    The return value is the pointer position after the walk.
-    [Arguments]    ${steps}=0    ${delay}=0
+Walk Pointer To (${x}, ${y})
+    [Documentation]    Move the pointer in incremental steps from the
+    ...    current pointer position to an absolute position.
+    ...
+    ...    Arguments:
+    ...    - ${x}: Integer absolute x-coordinate to walk the pointer to.
+    ...    - ${y}: Integer absolute y-coordinate to walk the pointer to.
+    ...    - ${steps} (optional): number of steps. If <= 0, the pointer is
+    ...      moved in increments of 16 pixels. Default is 0.
+    ...    - ${delay} (optional): time to sleep after each step, in seconds.
+    ...      Default is 0.01.
+    [Arguments]    ${steps}=0    ${delay}=0.01
```

Comment:
I'd move `[Arguments]` up near the keyword, if possible?

Reply:
Yes, done... although I'd still prefer the documentation to come first as a docstring. The `[Arguments]` tag in this case contains only the optional arguments.